azkaban-common/src/main/java/azkaban/utils/DirectoryFlowLoader.java

```diff
@@ -366,6 +376,27 @@ private void constructFlow(Flow flow, Node node, Set<String> visited) {
     visited.remove(node.getId());
   }
 
+  private void jobPropertiesCheck() {
+    String maxXms = props.getString(JOB_MAX_XMS, MAX_XMS_DEFAULT);
+    String maxXmx = props.getString(JOB_MAX_XMX, MAX_XMX_DEFAULT);
+    long sizeMaxXms = azkaban.utils.Utils.parseMemString2KB(maxXms);
+    long sizeMaxXmx = azkaban.utils.Utils.parseMemString2KB(maxXmx);
+
+    for (String jobName : jobPropsMap.keySet()) {
+      Props jobProps = jobPropsMap.get(jobName);
+      String xms = jobProps.getString(XMS, null);
+      if (xms != null && azkaban.utils.Utils.parseMemString2KB(xms) > sizeMaxXms) {
```

Comment:
Do we have properties resolved at this point ? Someone may have a 
XMS={FROM_FLOW} which will not be a valid value, if unresolved ?